doc/designs/ech-api.md

```diff
@@ -0,0 +1,590 @@
+Encrypted ClientHello (ECH) APIs
+================================
+
+There is an [OpenSSL fork](https://github.com/sftcd/openssl/tree/ECH-draft-13c)
+that has an implementation of Encrypted Client Hello (ECH) and these are design
+notes relating to the current APIs for that, and an analysis of how these
+differ from those currently in the boringssl library.
```

Comment:
yep, I'm assuming that since the feature branch will take some time to be fully functional, including this note in the meanwhile will remain useful, but it'll need replacing towards the end

Reply:
ok